src/stan_math_backend/Stan_math_code_gen.ml

```diff
@@ -631,36 +636,44 @@ let pp_unconstrained_param_names ppf {Program.output_vars; _} =
 
 (** Print the `transform_inits` method of the model class *)
 let pp_transform_inits ppf {Program.transform_inits; _} =
+  pf ppf
+    "template <typename VecVar, typename VecI,\n\
+    \     stan::require_std_vector_t<VecVar>* = nullptr,\n\
+    \     stan::require_vector_like_vt<std::is_integral, VecI>* = nullptr>@ " ;
   let params =
-    [ "const stan::io::var_context& context__"; "std::vector<int>& params_i__"
-    ; "std::vector<double>& vars__"; "std::ostream* pstream__" ]
+    [ "const stan::io::var_context& context__"; "VecI& params_i__"
+    ; "VecVar& vars__"; "std::ostream* pstream__ = nullptr" ]
   in
   let intro =
     [ "using local_scalar_t__ = double;"; "vars__.clear();"
     ; "vars__.reserve(num_params_r__);" ]
   in
-  let cv_attr = ["const"; "final"] in
-  pp_method_b ppf "void" "transform_inits" params intro transform_inits
+  let cv_attr = ["const"] in
+  pp_method_b ppf "void" "transform_inits_impl" params intro transform_inits
     ~cv_attr
 
 (** Print the `log_prob` method of the model class *)
 let pp_log_prob ppf Program.({prog_name; log_prob; _}) =
-  pf ppf "template <bool propto__, bool jacobian__, typename T__>@ " ;
+  pf ppf
+    "template <bool propto__, bool jacobian__, typename VecR, typename VecI,\n\
+    \     stan::require_vector_like_t<VecR>* = nullptr,\n\
+    \     stan::require_vector_like_vt<std::is_integral, VecI>* = nullptr>@ " ;
```

Comment:
Is there a reason these are red?

Reply:
Sorry, forgot about this Q. You should just let the formatter handle newlines. Adding manual \n leads to a war with the formatter in my experience :)